pkg/controller/daemon/controller.go

```diff
@@ -383,22 +384,35 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) {
 	dsc.expectations.SetExpectations(dsKey, len(nodesNeedingDaemonPods), len(podsToDelete))
 
 	glog.V(4).Infof("Nodes needing daemon pods for daemon set %s: %+v", ds.Name, nodesNeedingDaemonPods)
+	createWait := sync.WaitGroup{}
+	createWait.Add(len(nodesNeedingDaemonPods))
 	for i := range nodesNeedingDaemonPods {
-		if err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[i], ds.Namespace, ds.Spec.Template, ds); err != nil {
-			glog.V(2).Infof("Failed creation, decrementing expectations for set %q/%q", ds.Namespace, ds.Name)
-			dsc.expectations.CreationObserved(dsKey)
-			util.HandleError(err)
-		}
+		go func(ix int) {
```

Comment:
@mqliang look at how BurstReplicas is used in the replication controller

https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replication/replication_controller.go#L369-L371